README.md

```diff
@@ -1,208 +1,262 @@
-# Mendix Push notifications
+# Mendix Push Notifications
 
-This module and widget should be used to implement push notifications for android, ios and windows tablets/mobiles. This module/widget has been tested to work on Mendix 5.14.1 and up. 
+Push notifications let your application notify a user of events even when the user is not actively using the application. This is a native capability provided by both Android and iOS devices and made available via Google Cloud Messaging (GCM) and Apple Push Notifications service (APNs). This project is meant to make it easy for Mendix developers who want to include Push Notifications capability into their Mendix hybrid mobile application.
 
-The project contains:
+## Overview
 
-- Directory structure.
-- Readme.md file.
-- Javascript source.
-- XSD for package.xml, to configure properties of the widget, visible inside the Mendix business modeler.
-- Example project
-- Module for sending push notifications
-- Custom css file to style the push notification html
+<img src="assets/images/overview/architecture.png"/>
+
+In general, a Mendix Push Notifications solution consists of two parts: the MxPushNotifications module and the Push Notification widget. The module is the "server-side" and responsible for sending push notifications to GCM/APNs which in turn will send the notifications to end-user devices. The widget resides in the hybrid mobile (phonegap) app. In general, it has two responsibilities: to make GCM/APNs aware about the particular devices they run in, and to be able to handle push notifications coming from GCM/APNs.
+
+## Limitation
+
+This project assumes that the mobile app and the "back-end" part will be co-located in the same application.
+
+At this moment this project does not support:
+- Off-line hybrid mobile
```

Comment:
Is it actually common to use the dash in off-line? I usually just see offline being used.